### DISCUSSION TITLE
TanTan WP3 Button Not Working

### DISCUSSION THREAD

**Discussion by datrumole** [arendst/Tasmota#18087] [Q&A]
hey all, not sure if i should be opening a new issue for this or using discussions, but i figured its likely a 'me problem' not tasmota (at this time)

i have a tantan wp3, which is identical to the gosund wp3. after figuring out how to get to the rx/tx pins in the back of the unit, i got tasmota successfully flashed

literally just leaving the device as a 'sonoff basic' everything BUT the button works (wifi, mqtt, the led's switching between red/blue properly)

i found this page, where someone did a really good tear down of the schematics: https://vmallet.com/2020/07/gosund-wp3-smart-plug-teardown-and-schematic/ 

i have tried the tantan config, the gosund config (those introduce new issues as the led's dont work properly, and dont fix the button either), heck, even using the schematics in link above to attempt my own config. i have messed with the switchmode, i have tried pretty much every mode, its like the button doesnt even register 

i'm sure it's something i've done wrong, just dont know where to go from here at this point

**Comment by barbudor** [comment]
The schematics doesn't match any of the template
So difficult to say who is right as you or others may have a variation of the design
If anything but the button works, I suggest you move the Button_1 to any gpio that is not yet used and try until you find it

**Reply by datrumole** [reply]
thanks, this is going to be a dumb question, but just by finding the right GPIO for 'button1', will it automatically toggle the relay? or does something need to be told to

edit: also, once i move off the 'sonoff basic' i lose the led functionality. they either go off, red when on, both when on, or none on all all when toggled

**Reply by barbudor** [reply]
Button 1 is by default associated to Relay 1 so as soon as you have the right GPIO, it will toggle the relay
